docs: document replica scheduling constraints (#21090)
MaterializeInc/materialize: doc/user/content/sql/create-cluster.md

{{% replica-options %}}

> ## Availability Zones and Scheduling

Reviewer: ```suggestion
## Availability zone assignment
```

(I think "assignment" is a bit clearer for the end user, since they're not thinking about the Kubernetes scheduler.)

> When scheduling replicas, Materialize makes the following guarantees, within each cluster:

Reviewer: ```suggestion
When assigning replicas of a managed cluster to availability zones, Materialize makes the following guarantees within each cluster:

{{< warning >}}
These guarantees to do not apply to unmanaged clusters.
{{< /warning >}}
```

(There are _some_ guarantees, but I think it's too confusing to try to talk about them.)

Author: they do apply to unmanaged clusters though!

- Different replicas are _never_ scheduled on the same node.

> - Different replicas are _always_ spread evenly across availability zones.
>   - Replicas with more than 1 process are excluded from this constraint. See [`mz_internal.mz_cluster_replica_sizes`](/sql/system-catalog/mz_internal/#mz_cluster_replica_sizes)

Reviewer: ```suggestion
- Different replicas are _always_ spread evenly across availability zones. **Known limitation:** replicas with more than 1 process are excluded from this constraint. See [`mz_internal.mz_cluster_replica_sizes`](/sql/system-catalog/mz_internal/#mz_cluster_replica_sizes)
```

  to determine if that is the case for your replicas.

> - Replicas with a more than 1 process _prefer_ being placed in the same availability zones.
>   - This is because multi-process replicas are shared-fate.

Reviewer: I think this is a can of worms that we can keep the lid on! The rest of our docs talk about replicas as a unit. So I think we're safe here not to mention the possibility of skew between processes in a multi-process replica, because no one will think of that being a possibility!

```suggestion
```

Author: sounds good!

> - Replicas with a specific `AVAILABILITY ZONE` are _always_ scheduled in that availability zones.
>   - These replicas still participate in spread constraints.

Reviewer: I think we should just axe this and restrict our explanation to managed cluster replicas. It's sooo hard to explain how the replicas are considered in other replica's spread constraints but also pin themselves to their own AZ...

```suggestion
```
